#! /bin/sh /usr/share/dpatch/dpatch-run
## 03_gcc43.dpatch by Cyril Brulebois
## DP: Fix compilation with gcc-4.3.

@DPATCH@
--- a/lib/frei0r/filter/equaliz0r/equaliz0r.cpp
+++ b/lib/frei0r/filter/equaliz0r/equaliz0r.cpp
@@ -19,6 +19,8 @@
 
 #include "frei0r.hpp"
 
+#include <cstring>
+
 /* Clamps a int32-range int between 0 and 255 inclusive. */
 unsigned char CLAMP0255(int32_t a)
 {
--- a/lib/frei0r/filter/sobel/sobel.cpp
+++ b/lib/frei0r/filter/sobel/sobel.cpp
@@ -19,6 +19,8 @@
 
 #include "frei0r.hpp"
 
+#include <cstdlib>
+
 /* Clamps a int32-range int between 0 and 255 inclusive. */
 unsigned char CLAMP0255(int32_t a)
 {
--- a/lib/frei0r/generator/partik0l/partik0l.cpp
+++ b/lib/frei0r/generator/partik0l/partik0l.cpp
@@ -35,6 +35,8 @@
 
 #include <config.h>
 
+#include <cstring>
+
 
 /* defines for blob size and roundness */
 #define LIM 8 // 25
--- a/lib/tbt/tbt.cpp
+++ b/lib/tbt/tbt.cpp
@@ -28,6 +28,8 @@
 #include <tbt.h>
 #include <jutils.h>
 
+#include <cstring>
+
 
 
 
